### PR TITLE
add alter table only implementation

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: docs/sql/reference/alter_table.txt [hunk 1]
```diff
@@ -12,7 +12,8 @@ Synopsis
 
 ::
 
-    ALTER [BLOB] TABLE table_ident [ PARTITION (partition_column = value [ , ... ]) ]
+    ALTER [ BLOB ] TABLE [ [ ONLY ] table_ident
```
Comment (mfussenegger) [inline docs/sql/reference/alter_table.txt]: should this be ` { ONLY table_ident | table_ident [ PARTITION ...] }`?

File: sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java [hunk 2]
```diff
@@ -363,7 +363,11 @@ public void onFailure(Throwable t) {
         TableParameter concreteTableParameter = tableParameter;
         if (analysis.table().isPartitioned()) {
             if (analysis.partitionName().isPresent()) {
-                indices = new String[]{ analysis.partitionName().get().stringValue() };
+                indices = new String[]{analysis.partitionName().get().stringValue()};
+            } else if (analysis.excludePartitions()) {
+                updateTemplate = true;
+                updateMapping = false;
+                indices = new String[]{ analysis.table().ident().esName() };
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java]: shouldn't the indices be empty in this case?

File: sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java [hunk 3]
```diff
@@ -435,7 +438,9 @@ public void onFailure(Throwable e) {
             });
 
         }
-        if (!concreteTableParameter.settings().getAsMap().isEmpty() && indices.length > 0) {
+        if (!concreteTableParameter.settings().getAsMap().isEmpty()
+                && indices.length > 0
+                && !analysis.excludePartitions()) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java]: if the indices are empty (see above) the  excludePartition check here isn't required

File: docs/sql/partitioned_tables.txt [hunk 4]
```diff
@@ -306,6 +306,17 @@ the :ref:`ref-alter-table-partition-clause`.
     cr> ALTER TABLE parted_table PARTITION (day=1396915200000) RESET (number_of_replicas)
     ALTER OK (... sec)
 
+Alter Table ONLY
+----------------
+
+Sometimes one want to alter a partitioned table, but the changes
```
Comment (mfussenegger) [inline docs/sql/partitioned_tables.txt]: one wants to...

File: docs/sql/reference/alter_table.txt (hunk 1 above)
Comment (seut) [inline docs/sql/reference/alter_table.txt]: ah yeah right

**Comment by mfussenegger** [comment]
ack

**Pull request merged**